src/driver/mysql/MysqlDriver.ts

```diff
@@ -217,11 +217,17 @@ export class MysqlDriver implements Driver {
             case ColumnTypes.BOOLEAN:
                 return value === true ? 1 : 0;
             case ColumnTypes.DATE:
-                return moment(value).format("YYYY-MM-DD");
+                if (moment(value).isValid())
+                  return moment(value).format("YYYY-MM-DD");
+                else return '0000-00-00';
```

Comment:
the problem in new Date('0000-00-00').  See [here](https://github.com/typeorm/typeorm/pull/73/commits/146d1499d1cb55dc8c3323af2071cd6c5529daa0) my latest changes in this area